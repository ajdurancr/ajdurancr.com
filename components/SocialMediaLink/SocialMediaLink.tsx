import React from 'react'
import get from 'lodash.get'

import { Image } from '../Image'
import { getImageThemeSuffix } from '../../helpers/appTheme'
import useAppTheme from '../../hooks/useAppTheme'
import styles from './SocialMediaLink.module.sass'

type MediaLinkProps = {
  icon: string
  cta: {
    link: string
    text: string
  }
} 

const SocialMediaLink: React.FC<MediaLinkProps> = ({ cta, icon }) => {
  const appTheme = useAppTheme()

  if (!cta && icon) return null

  const imageSuffix = getImageThemeSuffix(appTheme)

  return (
    <a href={get(cta, 'link')} className={styles.link}>
      <Image
        src={`/images/${icon}${imageSuffix}.svg`}
        alt={get(cta, 'text')}
      />
    </a>
  )
}

export { SocialMediaLink }
