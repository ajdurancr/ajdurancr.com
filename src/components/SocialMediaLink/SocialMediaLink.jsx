import React from 'react'
import propTypes from 'prop-types'
import get from 'lodash.get'

import { Image } from '..'
import { getImageThemeSuffix } from '../../helpers/appTheme'
import useAppTheme from '../../hooks/useAppTheme'
import styles from './SocialMediaLink.module.sass'

const SocialMediaLink = ({ cta, icon }) => {
  const appTheme = useAppTheme()

  if (!cta && icon) return null

  const imageSuffix = getImageThemeSuffix(appTheme)

  return (
    <a href={get(cta, 'link')} className={styles.link}>
      <Image
        src={`/images/${icon}${imageSuffix}.svg`}
        alt={get(cta, 'text')}
      />
    </a>
  )
}

SocialMediaLink.propTypes = {
  icon: propTypes.string.isRequired,
  cta: propTypes.shape({
    link: propTypes.string.isRequired,
    text: propTypes.string.isRequired,
  }).isRequired,
}

export default SocialMediaLink
