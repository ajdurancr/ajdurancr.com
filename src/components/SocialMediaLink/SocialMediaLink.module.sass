.link
  display: block
  width: 3rem
  height: 3rem
  margin-right: .4rem
  