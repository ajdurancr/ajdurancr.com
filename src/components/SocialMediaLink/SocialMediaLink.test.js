import React from 'react'
import { shallow } from 'enzyme'

import SocialMediaLink from './SocialMediaLink'
import socialMediaLinkProps from '../../mocks/formatted/socialMediaLink'

describe('SocialMediaLink', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SocialMediaLink {...socialMediaLinkProps} />)

    expect(wrapper).toMatchSnapshot()
  })
})
