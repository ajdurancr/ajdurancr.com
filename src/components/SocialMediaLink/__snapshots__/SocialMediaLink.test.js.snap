// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaLink renders correctly 1`] = `
<a
  className="link"
  href="https://github.com/ajdurancr"
>
  <Image
    alt="GitHub"
    className=""
    src="/images/github-light.svg"
  />
</a>
`;
